### PR TITLE
Drop python2 support for XenAPI

### PR DESCRIPTION
**Pull request by stephenchengCloud**
Tested:
202272 Ring3 BST+BVT (One failed case due to a known issue)
202275 ToolStack Input Karma (passed)
202276 Storage BST (passed, one failed case due to environmental issue, re-run it in 4066324)
4066324 XFSSmokeTest

### CODE REVIEW

**Review by liulinC — APPROVED** [review]

File: python3/Makefile [hunk 1]
```diff
@@ -52,6 +52,11 @@ install:
 	$(IDATA) perfmon/perfmon.service $(DESTDIR)/usr/lib/systemd/system/perfmon.service
 	$(IPROG) perfmon/sysconfig-perfmon $(DESTDIR)/etc/sysconfig/perfmon
 
+# example/python
+	$(IDATA) examples/XenAPIPlugin.py $(DESTDIR)$(SITE3_DIR)/
+	sed -i 's/#!\/usr\/bin\/python/#!\/usr\/bin\/python3/' $(DESTDIR)$(SITE3_DIR)/XenAPIPlugin.py
```
Comment (gangj) [inline python3/Makefile:57]: Not needed now.

Comment (stephenchengCloud) [reply python3/Makefile]: Deleted.

**Review by gangj — APPROVED** [review]

File: python3/examples/pyproject.toml [hunk 2]
```diff
@@ -3,4 +3,4 @@ requires = ["setuptools >= 38.6.0", "setuptools_scm[toml]", "wheel"]
 build-backend = "setuptools.build_meta"
 
 [tool.setuptools_scm]
```
Comment (edwintorok) [inline python3/examples/pyproject.toml:5]: You will need to update setup.cfg and probably remove this line:
```
[bdist_wheel]
# This flag says that the code is written to work on both Python 2 and Python
universal=1
```

Comment (stephenchengCloud) [reply python3/examples/pyproject.toml:5]: > You will need to update setup.cfg and probably remove this line:
> ```
> [bdist_wheel]
> # This flag says that the code is written to work on both Python 2 and Python 3
> universal=1
> ```
Thanks, removed.

File: python3/examples/setup.cfg [hunk 3]
```diff
@@ -20,9 +20,3 @@ classifiers =
 [options]
 packages = find:
 python_requires = >=3.6.*, <4
```
Comment (psafont) [inline python3/examples/setup.cfg:22]: Pip doesn't like this line, and will refuse to build the package, instead use:
```suggestion
python_requires = >=3.6, <4
```

**Review by psafont — CHANGES_REQUESTED** [review]

File: python3/examples/Makefile
Comment (psafont) [inline python3/examples/Makefile:1]: Can the line with `dune clean` be removed as well? Dune isn't building the python package anymore

**Review by psafont — APPROVED** [review]

**Pull request merged**